client/app/services/project.service.js

```diff
@@ -104,5 +105,46 @@
             });
             return feature;
         }
+
+        /**
+         * Validate a candidate AOI.
+         * @param features to be validated {*|ol.Collection.<ol.Feature>|Array.<ol.Feature>}
+         * @returns {{valid: boolean, message: string}}
+         */
+        function validateAOI(features){
+
+            var validationObj = {valid:true,message:''}
```

Comment:
Please put items on new line for readability. The 'obj' suffix shouldn't be needed because we can already see it is an object. Maybe rename to validationResult?
var validationResult = {
valid: true,
message: ''
}